strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -246,7 +246,7 @@ def init_circuit(self, prog, **kwargs):
         # Tensor product of the weights.
         weights = kron_list(init_weights)
         # De-nest the means iterator.
-        means = np.array([[a for b in tup for a in b] for tup in mean_combs])
+        means = np.array([[a for b in tup for a in b] for tup in mean_combs], dtype=object)
```

Comment:
I think dtype should be complex to match what the backend expects.

Reply:
I agree! We should make the mean coefficients always complex.

Reply:
```suggestion
        means = np.array([[a for b in tup for a in b] for tup in mean_combs], dtype=complex)
```